src/bootstrap/bootstrap.py

```diff
@@ -185,9 +186,63 @@ def download_stage0(self):
             if not os.path.exists(tarball):
                 get("{}/{}".format(url, filename), tarball, verbose=self.verbose)
             unpack(tarball, self.bin_root(), match="cargo", verbose=self.verbose)
+            self.fix_executable(self.bin_root() + "/bin/cargo")
             with open(self.cargo_stamp(), 'w') as f:
                 f.write(self.stage0_cargo_rev())
 
+    def fix_executable(self, fname):
+        # If we're on NixOS we need to change the path to the dynamic loader
+
+        default_encoding = sys.getdefaultencoding()
+        try:
+            ostype = subprocess.check_output(['uname', '-s']).strip().decode(default_encoding)
```

Comment:
I was aware of `platform.system()` but I copied what was done [here](https://github.com/rust-lang/rust/blob/10f6a5c4431e09d355a0ba319a630e02a1e38361/src/bootstrap/bootstrap.py#L316) because I figured there must be a reason for it. This shouldn't break on windows because the `fix_executable` function just returns without doing anything if something goes wrong.